source/winUser.py

```diff
@@ -430,7 +430,11 @@ def keybd_event(*args):
 	return user32.keybd_event(*args)
 
 def mouse_event(*args):
-	return user32.mouse_event(*args)
+	# #8452: NVDA should ignore mouse events that are injected by itself.
+	# Import late to avoid circular import
+	import mouseHandler
+	with mouseHandler.ignoreInjection():
+		return user32.mouse_event(*args)

```

Comment:
I don't think that we can rely on the timing of receiving these events. Currently this design relies on the events being received while this contextManager is active.

Reply:
Hmmmm, a similar approach is used in keyboardHandler.KeyboardInputGesture.send, though there, wx.Yield is called before closing the context manager.

Reply:
I have a feeling that code relies on a sleep, which I'm not a fan of either! Really we need a way to identify that a future received event is the one we planned to ignore, and only ignore that one.

Reply:
Hmm yes, this could be a cause of #8020.
Note that similar to wx.Yield, we also have wx.YieldIfNeeded.
I"m still not sure what to do here. @michaeldcurran or @jcsteh, thoughts?

Reply:
I discovered that there's also `wx.GetApp().SafeYieldFor` which allows a bitmask of events to be provided, limiting the yield to whatever events you provide.

Regarding SafeYield:

> This function is similar to wx.Yield , except that it disables the user input to all program windows before calling wx.AppConsole.Yield and re-enables it again afterwards.